cmd/ncproxy/store.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/errdefs"
+	"github.com/pkg/errors"
+	bolt "go.etcd.io/bbolt"
+)
+
+type computeAgentStore struct {
+	DB *bolt.DB
+}
+
+func newComputeAgentStore(db *bolt.DB) *computeAgentStore {
+	return &computeAgentStore{DB: db}
+}
+
+func (n *computeAgentStore) get(ctx context.Context, key string) (result string, err error) {
+	if err := n.DB.View(func(tx *bolt.Tx) error {
+		bkt := getComputeAgentBucket(tx)
+		if bkt == nil {
+			return errors.Wrapf(errdefs.ErrNotFound, "Compute Agent bucket %v", bucketKeyComputeAgent)
+		}
+		data := bkt.Get([]byte(key))
+		if len(data) == 0 {
+			return errors.Wrapf(errdefs.ErrNotFound, "Compute Agent %v", key)
+		}
+		result = string(data)
+		return nil
+	}); err != nil {
+		return "", err
+	}
+
+	return result, nil
+}
+
+func (n *computeAgentStore) getKeyValueMap(ctx context.Context) (map[string]string, error) {
```

Comment:
I think a better name could be used here that more accurately describes what is being returned. Perhaps `getActiveComputeAgents` with a comment explaining what the map key and value represent?